00-basics/002-javascript/Kristina/Learning003.js

```diff
@@ -0,0 +1,64 @@
+var myArray = [["John", 23], ["dog", 3]];
+myArray.shift();
+myArray.unshift(["Paul",35])
+console.log(myArray);
+
+var myList = [
+    ["Chocolate Bar", 15],["Banana",25],["Dragonfruit",60],["Muesli",17],["Rainbow skittles",365]
+];
+console.log(myList);
+
+function reusableFunction(){
+    console.log("Hi World");
+};
+reusableFunction();
+
+function functionWithArgs (param1, param2){
+    console.log(param1+param2);
+}
+functionWithArgs(1,2)
+
+function myLocalScope() {
+    var myVar="Test"
+    console.log('inside myLocalScope', myVar);
+}
+myLocalScope();
+//console.log('outside myLocalScope', myVar);
+//^This is supposed to not work, and does not work
+
+var outerWear = "T-Shirt";
+function myOutfit() {
+    var outerWear="sweater"
+    return outerWear;
+}
+myOutfit();
+//local takes precedence over global variable here^
+
+var sum = 0;
+function addThree() {
+    sum = sum + 3;
+}
+function addFive(){
+    sum+=5
+}
+console.log(addThree());
+console.log(addThree(addFive));
+//^why functions have return statements
```

Comment:
Is this a question or explanation to the block in question?

In case you were asking, `return` means that you function can be used either for setting a value, or for passing it to another function:

```
const x = myFunction()
anotherFunction(myFunction())
```

If you do not define `return` value inside the function, it returns `undefined`.

Alternatively, as you see in your example above, you can use functions to set outside values directly (`sum = <...>`). Functions that do that are called `impure` function that have `side effects` and they are **very** discouraged, because they make code harder to follow.